### PR TITLE
Fixes and improvements for Iceberg storage

### PR DESCRIPTION
**Pull request by Avogar**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Throw an exception when Iceberg table schema was evolved instead of possible incorrect result. Throw an exception if positional or equality deletes were used in Iceberg table instead of possible incorrect result. Fix incorrect metadata file selection (lexicographic order was used and we could select `v9.metadata.json` instead of `v10.metadata.jsonl`). Use schema from metadata file for schema inference instead of listing data files and inferring schema from files. Don't list data files on each read if metadata wasn't changed. Fix parsing Iceberg decimals and timestamptz for ORC and Avro formats.

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by Avogar** [comment]
TODO: add tests and docs about unsupported Iceberg features

File: src/Storages/DataLakes/Iceberg/IcebergMetadata.cpp
Comment (Avogar) [inline src/Storages/DataLakes/Iceberg/IcebergMetadata.cpp:1]: Half of the code is not new and was copied from `IcebergMetadataParser`.

File: src/Storages/DataLakes/Iceberg/IcebergMetadata.cpp [hunk 1]
```diff
@@ -0,0 +1,514 @@
...
+
+            content_int_column = assert_cast<const ColumnInt32 *>(content_column.get());
+        }
+
+        for (size_t i = 0; i < tuple_column->size(); ++i)
+        {
+            if (format_version == 2)
+            {
+                Int32 content_type = content_int_column->getElement(i);
+                if (content_type != 0)
```
Comment (kssenii) [inline src/Storages/DataLakes/Iceberg/IcebergMetadata.cpp:497]: a comment could be useful
```suggestion
                /// 0=data, 1=position deletes, 2=equality deletes; default to 0 when reading v1 manifests
                if (content_type != 0)
```

Comment (Avogar) [reply src/Storages/DataLakes/Iceberg/IcebergMetadata.cpp]: I'd better add a enum for all content types with the same values

**Review by kssenii — APPROVED** [review]

**Comment by Avogar** [comment]
Users faced issues in cloud with Iceberg, let's backport this PR.

**Pull request merged**